src/c/soubory/prace_se_soubory.md

```diff
@@ -0,0 +1,240 @@
+jeden prvek a parameter `count` nastavit na celkovou velikost všech dat, které chceme zapsat:
+```c
+int pole[5] = { 1, 2, 3, 4, 5 };
+fwrite(pole, sizeof(pole), 1, soubor);
+```
+
+[^2]: V takovémto případě by funkce `fwrite` vrátila hodnotu `3`.
+
+Pokud bychom zapsali `pole` do souboru takto, uloží se do něj celkem `20` (`5` * `4`) bytů (čísel),
+které později můžeme v programu zase [načíst zpátky](#čtení-ze-souboru). Pokud bychom se podívali,
+co v souboru je, nalezli bychom seznam čísel `1 0 0 0 2 0 0 0 3 0 0 0 4 0 0 0 5 0 0 0`, což odpovídá
+paměťové reprezentaci pole pěti `int`ů, které bylo vytvořeno výše.
+
+### Textový zápis
+Pokud bychom si tato data chtěli přečíst jako text, můžeme čísla z výše zmíněného pole zapsat do
+souboru pomocí nějakého textového kódování, například [ASCII](../text/znaky.md). K tomu můžeme
+využít funkci [`fprintf`](https://devdocs.io/c/io/fprintf), která funguje stejně jako `printf`, s
+tím rozdílem, že text nevypisuje na `stdout`, ale do předaného souboru:
+```c
+#include <stdio.h>
+#include <assert.h>
+
+int main() {
+    int pole[5] = { 1, 2, 3, 4, 5 };
+
+    // otevření souboru
+    FILE* soubor = fopen("soubor", "wb");
+    assert(soubor);
+
+    // zápis do souboru
+    int zapsano = fwrite(pole, sizeof(int), 5, soubor);
+    assert(zapsano == 5);
+
+    // zavření souboru
+    fclose(soubor);    
+
+    return 0;
+}
+```
+V tomto případě by se do souboru zapsalo deset bytů (čísel) `49 32 50 32 51 32 52 32 53 32`,
+které bychom mohli např. pomocí textového editoru interpretovat jako
+[ASCII](https://www.asciitable.com/) text `1 2 3 4 5 `.
+
+### Bufferování
+Stejně jako při zápisu do `stdout` se i při zápisu do souborů uplatňuje
+[*bufferování*](../text/vstupavystup.md#standardní-souborové-deskriptory). Data, která do souboru
+zapíšeme, se tak v něm neobjeví hned. Pokud bychom chtěli donutit náš program, aby data uložená
+v bufferu opravdu vypsal do souboru, můžeme použít funkci [`fflush`](https://devdocs.io/c/io/fflush)
+[^3].
+
+[^3]: Ani zavolání funkce `fflush` však nezajistí, že se data opravdu zapíšou na fyzické médium
+(například harddisk). To je ve skutečnosti velmi obtížný [problém](https://lwn.net/Articles/457667/).
+
+## Čtení ze souboru
+Pro čtení ze souboru můžeme použít funkci [`fread`](https://devdocs.io/c/io/fread), která je
+protikladem funkce `fwrite`:
+```c
+size_t fread(
+    void* buffer,   // adresa, na kterou zapíšeme data ze souboru
+    size_t size,    // velikost prvku, který načítáme
+    size_t count,   // počet prvků, které načítáme
+    FILE* stream    // soubor, ze kterého čteme
+);
+```
+Tato funkce opět předpokládá, že budeme ze souboru načítat několik hodnot stejného datového typu.
+Například načtení pěti celých čísel, které jsme zapsali v kódu [výše](#zápis-do-souboru), by mohlo
+vypadat následovně:
+```c
+#include <stdio.h>
+#include <assert.h>
+
+int main() {
+    int pole[5] = { 1, 2, 3, 4, 5 };
+
+    // otevření souboru
+    FILE* soubor = fopen("soubor", "rb");
```

Comment:
Mozna bych pouzil priponu `.bin`, at se jasne oddeli textove a binarni soubory.

Reply:
Mi právě přišlo lepší ukázat, že to příponu třeba ani mít nemusí. K textovému `.txt` dám.